### PR TITLE
Feature: Ordnance Survey map

### PR DESCRIPTION
**Pull request by trovster**
Looking at implementing OS map with Mapbox GL. However, there is an issue with `countLayer` configuration;

```js
{
  'text-font': ['Open Sans Bold'],
  'text-field': '{point_count_abbreviated}',
  'text-size': 16
}
```

When removing the `text-field` value, the map renders albeit without the cluster marker number. Replacing this with the `markerLayer` displays correctly, with the marker within the cluster.

I have researched this and found out about required font glyphs which talk about solving an issue but doesn't seem to resolve this one. See https://github.com/mapbox/mapbox-gl-js/issues/4808

### CODE REVIEW

**Comment by trovster** [comment]
I have investigated this a lot and can't find a solution. Maybe it's worth fresh eyes. I have written up my notes in this PR.

**Comment by madwort** [comment]
Rebased to add bugfixes from develop branch

**Comment by madwort** [comment]
Hmmm, been trying to figure out what's going wrong - what I'm currently seeing is these "Unimplemented type 3" messages:
Screenshot 2021-03-05 at 17 03 43
which appear to correspond to these apparently successful requests (`200 OK`) for a font file from the OS server:
Screenshot 2021-03-05 at 17 04 10

There's lots of mentions of this error being due to the tileserver sending gzipped content without the gzip header (so the JS throws this error when it fails to parse it) - but these font requests appear to have the gzip header set (which should tell the browser to decode it before giving the content to JS).

**Comment by madwort** [comment]
Ah - it's actually a `200 OK` containing a `404` json response!

```
curl -D - 'https://api.os.uk/maps/vector/v1/vts/resources/fonts/Open%20Sans%20Bold/0-255.pbf?key=rG5uUoghjuU8diBdnOZTnUBgOhxJuvTh&srs=3857'
HTTP/1.1 200 OK
Date: Fri, 05 Mar 2021 16:58:06 GMT
Content-Type: application/json;charset=utf-8
Content-Length: 88
Connection: keep-alive
X-XSS-Protection: 1; mode=block
X-Content-Type-Options: nosniff
Via: 1.1 ace508199bd84d41add2e7c3b2f8dada.cloudfront.net (CloudFront)
Vary: Origin,Accept-Encoding,key
OMSE-CATEGORY: open
OMSE-TRANSACTION-COUNT: 0
OMSE-PREMIUM-COUNT: 0
Access-Control-Allow-Origin: *
Access-Control-Allow-Credentials: true
Access-Control-Allow-Headers: origin, x-requested-with, accept, key, authorization, content-type
Access-Control-Max-Age: 3628800
Access-Control-Allow-Methods: GET, POST
Strict-Transport-Security: max-age=31536000; includeSubDomains

{"error":{"code":404,"message":"Requested font resource not available.","details":null}}%        
```

**Comment by madwort** [comment]
Just been trying to tell mapbox-gl to load the font from a different location (mapbox / https://github.com/geops/mapbox-gl-fonts / etc) but no dice yet.

**Comment by madwort** [comment]
Here it is working:
Screenshot 2021-03-05 at 17 23 15
(nb. OS branding added separately)

File: src/components/utilities/map/js/config/map-os.js [hunk 1]
```diff
@@ -3,7 +3,6 @@ const { osApiKey } = require('./env')
 const baseUrl = 'https://api.os.uk/maps/vector/v1/vts'
 
 module.exports = {
-  glyphs: 'mapbox://fonts/openmaptiles/{fontstack}/{range}.pbf',
```
Comment (madwort) [inline src/components/utilities/map/js/config/map-os.js]: this seems to be ignored!

File: src/components/utilities/map/js/config/countLayer.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 module.exports = {
-  'text-font': ['Open Sans Bold'],
+  'text-font': ['Arial Bold'],
```
Comment (madwort) [inline src/components/utilities/map/js/config/countLayer.js:2]: `Open Sans Bold` not present on the OS server, `Arial Bold` is.

**Comment by trovster** [comment]
Is this working now? Basically you need to find a supported font?

**Comment by madwort** [comment]
Yep it's working - it seems to be insisting on getting the font from the Ordnance Survey api, so it needs to be one of their supported fonts. Ideally I guess we'd figure out how to get the font from somewhere else & use a better font, but it's sort of unblocked for now.

**Pull request merged**